### ISSUE TITLE
Calling `enableAutomaticMode` destroys all tracing in Lambda environment

### ISSUE THREAD

**Issue by gavinpc-mindgrub** [aws/aws-xray-sdk-node#366]
X-Ray's "Manual mode" is not supported in Lambda environments, and "Automatic Mode" is enabled by default.  So there is no reason to call `enableAutomaticMode`.

However, if you _do_ call `enableAutomaticMode()` from a Lambda environment, bad things will happen:

- it will raise a "context missing" error
- if your "context missing" strategy is to log, your code will continue running, but the "facade" segment will never be assigned
- and therefore every subsequent attempt to trace will also raise "context missing", filling your logs
- you may also spend time looking at every possible cause except for the call to `enableAutomaticMode`, which seems innocuous since it should be a no-op.

In Lambda environments, [the `enableManualMode` function is overridden](https://github.com/aws/aws-xray-sdk-node/blob/master/packages/core/lib/env/aws_lambda.js#L20) to issue a warning:

```
  contextUtils.enableManualMode = function() {
    logger.getLogger().warn('AWS Lambda does not support AWS X-Ray manual mode.');
  };
```

I suggest that the same should be done for `enableAutomaticMode`, to prevent new users from losing a lot of time over it (as I just did).

**Comment by willarmiros** [comment]
@gavinpc-mindgrub I've opened #371 to address this, thanks!

**Comment by willarmiros** [comment]
Released in 3.3.0
